MapboxStatic/Snapshot.swift

```diff
@@ -503,49 +503,52 @@ public class Snapshot: NSObject {
             switch response.statusCode {
             case 429:
-                if let timeInterval = response.allHeaderFields["x-rate-limit-interval"] as? NSTimeInterval, maximumCountOfRequests = response.allHeaderFields["x-rate-limit-limit"] as? UInt {
-                    let intervalFormatter = NSDateComponentsFormatter()
-                    intervalFormatter.unitsStyle = .Full
-                    let formattedInterval = intervalFormatter.stringFromTimeInterval(timeInterval)
-                    let formattedCount = NSNumberFormatter.localizedStringFromNumber(maximumCountOfRequests, numberStyle: .DecimalStyle)
+                if let timeInterval = response.allHeaderFields["x-rate-limit-interval"] as? TimeInterval, let maximumCountOfRequests = response.allHeaderFields["x-rate-limit-limit"] as? UInt {
+                    let intervalFormatter = DateComponentsFormatter()
+                    intervalFormatter.unitsStyle = .full
+                    let formattedInterval = intervalFormatter.string(from: timeInterval)
+                    let formattedCount = NumberFormatter.localizedString(from: NSNumber(value: maximumCountOfRequests), number: .decimal)
                     failureReason = "More than \(formattedCount) requests have been made with this access token within a period of \(formattedInterval)."
                 }
                 if let rolloverTimestamp = response.allHeaderFields["x-rate-limit-reset"] as? Double {
-                    let date = NSDate(timeIntervalSince1970: rolloverTimestamp)
-                    let formattedDate = NSDateFormatter.localizedStringFromDate(date, dateStyle: .LongStyle, timeStyle: .FullStyle)
+                    let date = Date(timeIntervalSince1970: rolloverTimestamp)
+                    let formattedDate = DateFormatter.localizedString(from: date, dateStyle: .long, timeStyle: .full)
                     recoverySuggestion = "Wait until \(formattedDate) before retrying."
                 }
             default:
                 failureReason = json["message"] as? String
             }
-            userInfo[NSLocalizedFailureReasonErrorKey] = failureReason ?? userInfo[NSLocalizedFailureReasonErrorKey] ?? NSHTTPURLResponse.localizedStringForStatusCode(error?.code ?? -1)
-            userInfo[NSLocalizedRecoverySuggestionErrorKey] = recoverySuggestion ?? userInfo[NSLocalizedRecoverySuggestionErrorKey]
+            userInfo[NSLocalizedFailureReasonErrorKey] = failureReason ?? HTTPURLResponse.localizedString(forStatusCode: statusCode)
+            userInfo[NSLocalizedRecoverySuggestionErrorKey] = recoverySuggestion ?? ""
         }
-        userInfo[NSUnderlyingErrorKey] = error
-        return NSError(domain: error?.domain ?? MBStaticErrorDomain, code: error?.code ?? -1, userInfo: userInfo)
+        userInfo[NSUnderlyingErrorKey] = error?.localizedDescription ?? ""
+        
+        return NSError(domain:MBStaticErrorDomain, code: -1, userInfo: userInfo)
```

Comment:
This discards the more specific domain and code that may potentially come from the URL response.